tasks/Logger.cpp

```diff
@@ -67,37 +67,45 @@ bool Logger::startHook()
       return false;
     }
 
+    _current_file.set(_file.value());
+
     if(_overwrite_existing_files.get() && _auto_rename_existing_files.get())
     {
       log(Error) << "Ambiguous property values. Both _overwrite_existing_files and _auto_rename_existing_files are set to true." << endlog();
       return false;
     }
 
-    if(boost::filesystem::exists(_file.value()) && !_overwrite_existing_files.get() && _auto_rename_existing_files.get())
+    if(boost::filesystem::exists(_current_file.get()) && !_overwrite_existing_files.get() && !_auto_rename_existing_files.get())
+    {
+      log(Error) << "File " << _current_file.get() << " already exists. No overwrite allowed by task property." << endlog();
+      return false;
+    }
+
+    if(boost::filesystem::exists(_current_file.get()) && !_overwrite_existing_files.get() && _auto_rename_existing_files.get())
```

Comment:
This actually makes the overall behavior inconsistent. The first file will never have a timestamp, only the following ones.

I'd suggest to renaming the property to _auto_timestamp_files and always add a timestamp when it is set.